src/lib/languages/javascript.rs

```diff
@@ -0,0 +1,144 @@
+use super::*;
+
+pub struct JavaScript {
+  pub src: String
+}
+
+impl JavaScript {
+  // Generate javascript code using tokens from parser
+  pub fn generate(parser: Parser) -> Result<Self, ParsingError> {
+    let mut code = Self{
+      src: String::new()
+    };
+    // define functions
+    for func in parser.functions {
+      let function = code.function(func);
+      code.src += &function;
+    }
+    for glob in parser.global_vars {
+      let global = code.global(glob);
+      code.src += &global;
+    }
+    // Because JS does not call main, we must do that here
+    code.src += "main();\n";
+    Ok(code)
+  }
+  pub fn function(&mut self, func: Function) -> String {
+    let mut src = String::new();
+    src += "function ";
+    src += &func.name.unwrap();
+    src += "(";
+    let mut index = 0;
+    while index < func.args.len() {
```

Comment:
I think you can better use one of: 
*I would prefer this one personally*
```rust
for arg in func.args.iter() {
````

or
```rust
for arg in &func.args {
```

or use the `ref` if the above 2 are not possible, i have seen it used in some places but haven't used it myself